catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DavEntry.java

```diff
@@ -0,0 +1,317 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+
+import com.github.sardine.DavResource;
+import com.github.sardine.Sardine;
+
+/**
+ * A webdav implementation of the {@link org.apache.commons.io.monitor.FileEntry}
+ * Uses https://github.com/lookfirst/sardine
+ */
+public class DavEntry implements Serializable {
+
+    private static final long serialVersionUID = -2505664948818681153L;
+
+    private static final DavEntry[] EMPTY_ENTRIES = new DavEntry[0];
+
+    private final DavEntry parent;
+
+    private DavEntry[] children;
+
+    private File file;
+
+    private String location;
+
+    private boolean exists;
+
+    private boolean directory;
+
+    private long lastModified;
+
+    private long length;
+
+    private String eTag;
+
+    /**
+     * @param location must be fully qualified
+     */
+    DavEntry(final String location) {
+        this(null, location);
+    }
+
+    private DavEntry(final DavEntry parent, final String location) {
+        if (location == null) {
+            throw new IllegalArgumentException("File is missing");
+        }
+        this.parent = parent;
+        this.setLocation(getLocation(location, parent));
+    }
+
+    static DavEntry[] getEmptyEntries() {
+        return EMPTY_ENTRIES;
+    }
+
+    boolean refresh(DavResource davResource) {
+
+        // cache original values
+        final boolean origExists = isExists();
+        final long origLastModified = getLastModified();
+        final boolean origDirectory = isDirectory();
+        final long origLength = getLength();
+        final String origEtag = geteTag();
+
+        // refresh the values
+        setExists(davResource != null);
+        setDirectory(isExists() && davResource.isDirectory());
+        setLastModified(isExists() && davResource.getModified() != null ?
+                davResource.getModified()
+                        .getTime() :
+                0);
+        setLength(isExists() && !isDirectory() ? davResource.getContentLength() : 0);
+        seteTag(isExists() ? davResource.getEtag() : "0");
+
+        if (!isDirectory() && (file != null || origLastModified != getLastModified())) {
+            FileUtils.deleteQuietly(file);
+        }
+
+        // Return if there are changes
+        return isExists() != origExists || getLastModified() != origLastModified
```

Comment:
✏️ Might be good to enforce more readable formatting here with EOL comments. Something like...
```java
return isExists() != origExists \\
        || getLastModified() != origLastModified \\
        || ...
```
That should make the formatter leave it alone.